Automattic/newspack-theme
# Set Up A Build Pipeline

### All Submissions:






### Changes proposed in this Pull Request:

This PR sets up a build pipeline based on `@automattic/calypso-build`, similar to the one used in Newspack Plugin, Newspack Blocks,  etc. This adds a new npm script, `build:webpack`, which will be run as part of `npm run build`. The script will transpile all scripts found in `js/src` and create identically named files in `js/dist`. All enqueuing paths have been updated. The one exception is `skip-link-focus-fix.js`, which is already minified and hard-coded in a template here: https://github.com/Automattic/newspack-theme/blob/master/functions.php#L348-L350. For simplicity's sake we'll leave this as-is. Dotorg requires including the source of minified code in the theme, so this has been moved to `js/originals`. Since it's not in `js/src`, Webpack will ignore it. 

In future PRs we can modernize the Javascript source, take advantage of WordPress components and libraries as needed, and consider combining multiple scripts in a single file. For now, this PR is purely about setting up the pipeline. 

### How to test the changes in this Pull Request:

1. Check out the branch. Run `npm install`. Run `npm run build`
2. Verify that `js/dist` folder is created. For every script in `js/src` there should be a corresponding file in `js/dist`as well as a dependencies json file. 
3. Verify that all scripts are loading and functioning as they did previously .

### Other information:
